### DISCUSSION TITLE
Feature Request: Tax Rules loaded by URL tag

### DISCUSSION THREAD

**Discussion by fuchsbart** [PrestaShop/PrestaShop#41074] [Ideas]
#### Is your feature request related to a problem?

Prestashop's tax rules work very well for many use cases. However, they are not applied until the customer's address is known - this moment happens usually at checkout. Until then, only the default tax rate can be shown to the potential buyer.

With the new EU taxation OSS procedure in place since July 2021, this limitation means that all customers shopping from a different EU country than the Prestashop merchant **will be shown an incorrect price**!
(If the tax rate of the seller and buyer from abroad doesn't coincidentally happen to be the same).

Another problem is that price comparison sites like Google Shopping perform a check of the reported price for different sales countries. With Prestashop, unfortunately, it is not possible to generate a URL per sales country that brings up the correct price. So Google gradually disables any offer reported for a different sales country than the home country - because even if the tax rule is configured correctly and a module with IP detection is used: Google's crawlers never come from the sales country for which the price was reported and therefore get a wrong price.

#### Describe the solution you'd like

The following feature would make Prestashop extremely valuable for all merchants in the EU:

A tag/query in the URL, similar to the language code, that loads the appropriate tax rule for the sales country and displays the correct price.

#### Alternatives you've considered

It is not possible to link the tax rule to a language. Neither does any EU merchant want to translate his store into all languages of the EU, nor can a separate language even be created for each individual country (with its own tax rates).

There are modules that ask the customer to select the country of delivery with a pop-up when the store is opened. However, in the age of cookie banners, it's not a solution for merchants who don't want to annoy their customers. 

Besides, module developers also only have the option of saving the value in the cookie or creating a guest address with the country in order to output the correct price. Cookies are not a solution for the problem with price comparisons like Google Shopping, and saving one address for each visitor increases the database for any slightly larger merchant with traffic.

## One Stop Shop
_The biggest reform of the VAT for EU online trade applies to every store operator in the EU since 01.07.2021_

- The previous delivery thresholds of the individual EU countries (EUR 35,000 or EUR 100,000) for VAT abroad when selling to private customers no longer apply.
- Instead, a Europe-wide delivery threshold of EUR 10,000 (net) for all EU countries in total per year has applied since 01.07.2021.
- This means that the vast majority of online retailers will be liable to pay tax in every EU country.
- The regular VAT reports that are thus required can be handled centrally for all EU countries via the One Stop Shop (OSS) of the country in which the company is based.
- For the VAT rate to be levied, it is decisive to which country sales are made - the tax rate applicable there for the products sold applies!

More information: [https://ec.europa.eu/taxation_customs/business/vat/vat-e-commerce/oss_en](https://ec.europa.eu/taxation_customs/business/vat/vat-e-commerce/oss_en)

**Comment by 0dp** [comment]
To add to this, sellers in EU that has to ship goods outside of EU are required to provide HS code and Country of Origin on each item line on the invoice and packing slip.

I also believe the above is a requirement of EU competition policies; that prices are not different from each EU state. Which they will appear as, the way it is now.

In general for all e-com platforms, not just prestashop. the whole tax/shipping system needs an overhaul.

 #18777

**Comment by hibatallahAouadni** [comment]
Hello @fuchsbart 

Thank you for your suggestion.
> A tag/query in the URL, similar to the language code, that loads the appropriate tax rule for the sales country and displays the correct price.

It seems a good idea but kind of risky, let's check it with the Product Team! 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

PS: thanks @0dp for your quick reply :wink: 

Thank you
